File: trip-kit-booking-ui/src/main/java/com/skedgo/android/tripkit/booking/ui/OAuth2CallbackHandler.java
package com.skedgo.android.tripkit.booking.ui;

import android.net.Uri;

import com.skedgo.android.tripkit.booking.BookingForm;

import rx.Observable;

public interface OAuth2CallbackHandler {
  Observable<BookingForm> handleOAuthURL(BookingForm bookingForm, Uri uri, String callback);
  Observable<BookingForm> handleRetryURL(BookingForm bookingForm, Uri uri);
}

File: trip-kit-booking-ui/src/main/java/com/skedgo/android/tripkit/booking/ui/OAuth2CallbackHandler.kt
package com.skedgo.android.tripkit.booking.ui

import android.net.Uri
import com.skedgo.android.tripkit.booking.BookingForm
import io.reactivex.Observable

interface OAuth2CallbackHandler {
  fun handleOAuthURL(bookingForm: BookingForm, uri: Uri, callback: String): Observable<BookingForm>
  fun handleRetryURL(bookingForm: BookingForm, uri: Uri): Observable<BookingForm>
}